include/SDL3_shadercross/SDL_shadercross.h

```diff
@@ -64,16 +64,42 @@ typedef struct SDL_ShaderCross_ComputePipelineInfo
     Uint32 threadCountX;                 /**< The number of threads in the X dimension. */
     Uint32 threadCountY;                 /**< The number of threads in the Y dimension. */
     Uint32 threadCountZ;                 /**< The number of threads in the Z dimension. */
-} SDL_ShaderCross_ComputePipelineInfo;
+} SDL_ShaderCross_ComputePipelineMetadata;
+
+typedef struct SDL_ShaderCross_SPIRV_Info
+{
+    const Uint8 *bytecode;                    /**< The SPIRV bytecode. */
+    size_t bytecodeSize;                      /**< The length of the SPIRV bytecode. */
+    const char *entrypoint;                   /**< The entry point function name for the shader in UTF-8. */
+    SDL_ShaderCross_ShaderStage shaderStage;  /**< The shader stage to transpile the shader with. */
+    bool enableDebug;                         /**< Allows debug info to be emitted when relevant. Can be useful for graphics debuggers like RenderDoc. */
+    const char *name;                         /**< A UTF-8 name to associate with the shader. Optional, can be NULL. */
+
+    SDL_PropertiesID props;                   /**< A properties ID for extensions. Should be 0 if no extensions are needed. */
+} SDL_ShaderCross_SPIRV_Info;
+
+typedef struct SDL_ShaderCross_HLSL_Info
+{
+    const char *hlslSource;                   /**< The HLSL source code for the shader. */
+    const char *entrypoint;                   /**< The entry point function name for the shader in UTF-8. */
+    const char *includeDir;                   /**< The include directory for shader code. Optional, can be NULL. */
+    char **defines;                           /**< An array of define strings. Optional, can be NULL. */
+    Uint32 numDefines;                        /**< The number of strings in the defines array. */
+    SDL_ShaderCross_ShaderStage shaderStage;  /**< The shader stage to compile the shader with. */
+    bool enableDebug;                         /**< Allows debug info to be emitted when relevant. Can be useful for graphics debuggers like RenderDoc. */
+    const char *name;                         /**< A UTF-8 name to associate with the shader. Optional, can be NULL. */
+
+    SDL_PropertiesID props;                   /**< A properties ID for extensions. Should be 0 if no extensions are needed. */
```

Comment:
Windows creation has an easy entry point (`SDL_CreateWindow`) and an "advanced mode" (`SDL_CreateWindowFromProperties`).
Can shadercross do the same? I'd like to avoid properties and struct member values to clash.

Reply:
Maybe, but we'd be almost doubling the API surface which is pretty cumbersome.